### ISSUE TITLE
[CST] [ENG] Remove the feature flag `cst_use_claim_details_v2` from vets-website

### ISSUE THREAD

**Issue by pmclaren19** [department-of-veterans-affairs/va.gov-team#87707]
## Background Context
Remove the feature flag `cst_use_claim_details_v2` from the Claim Status Tool and remove components that are no longer being used.

- 12 Components/Files have the feature flag and will need to be updated to always have the feature enabled
- 7 Test files have the feature flag and will need to be updated to always have the feature enabled
- 14 Components/Files need to be removed
- 14 Test files need to be removed

## Engineering 
**Components that we no longer need and can be removed**
- DetailsPage.jsx
- AddFilesFormOld.jsx
- NeedFilesFromYou.jsx
- ClaimsDecision.jsx
- ClaimComplete.jsx
- RequestedFilesInfo.jsx
- SubmittedTrackedItem.jsx
- AdditionalEvidenceItem.jsx
- AdditionalEvidencePageOld.jsx
- AddFilesFormOld.jsx
- EvidenceWarning.jsx
- FilesNeededOld.jsx
- FilesOptionalOld.jsx
- DueDateOld.jsx

**Test Files that we no longer need and can be removed**
- DetailsPage.unit.spec.jsx
- AddFilesFormOld.unit.spec.jsx
- NeedFilesFromYou.unit.spec.jsx
- ClaimsDecision.unit.spec.jsx
- ClaimComplete.unit.spec.jsx
- RequestedFilesInfo.unit.spec.jsx
- SubmittedTrackedItem.unit.spec.jsx
- AdditionalEvidenceItem.unit.spec.jsx.jsx
- AdditionalEvidencePageOld.unit.spec.jsx
- AddFilesFormOld.unit.spec.jsx
- EvidenceWarning.unit.spec.jsx
- FilesNeededOld.unit.spec.jsx
- FilesOptionalOld.unit.spec.jsx
- DueDateOld.jsx

**Components/ Files with feature flag**
- feature-toggle-5103-update-enabled.json
- feature-toggle-5103-update-enabled-v2.json
- feature-toggle-claim-detail-v2-enabled.json
- feature-toggle-claim-phases-enabled.json
- TrackClaimsPage.js
- TrackClaimsPageV2.js
- index.js
- routes.jsx
- TabNav.jsx
- DefaultPage.jsx
- ClaimStatusPage.jsx
- FilesPage.jsx

**Tests with feature flag:**
- ClaimDetailLayout.unit.spec.jsx
- ClaimStatusPage.unit.spec.jsx
- FilesPage.unit.spec.jsx
- TabNav.unit.spec.jsx
- DefaultPage.unit.spec.jsx
- 05.claim-document-request.cypress.spec.js
- index.unit.spec.js

## Out of scope
- Add here

## Open questions
- Add here

## Tasks
- [x] Remove 14 files
- [x] Remove 14 tests
- [x] Remove feature flag `cst_use_claim_details_v2` from 7  test files
- [x] Remove feature flag `cst_use_claim_details_v2` from 12  components/files
- [x] Remove feature flag `cst_use_claim_details_v2` from 7  test files
- [ ] Remove feature flag `cst_use_claim_details_v2` from featureFlagNames.json

## Definition of Done
- [ ] Meets acceptance criteria
- [ ] Passed E2E testing (90% coverage)
- [ ] Passed unit testing (90% coverage)
- [ ] Passed integration testing (if applicable)
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Team approved production verification process
- [ ] Design performs design QA and verifies the implementation matches the design spec
- [ ] Accessibility specialist performs accessibility review (in code or design)
- [ ] Engineering identifies staging users required to test and shares account and credentials with design and product
- [ ] Product performs functional QA and verifies acceptance criteria was met

## Acceptance Criteria
- [ ] The feature flag `cst_use_claim_details_v2` is no longer used in the vets-website CST
- [ ] Components that were only used when the feature flag `cst_use_claim_details_v2` were disabled are removed
- [ ] Tests that were only used when the feature flag `cst_use_claim_details_v2` were disabled are removed

**Comment by pmclaren19** [comment]
Reviewed during eng refinement.

**Comment by jerekshoe** [comment]
[Second PR](https://github.com/department-of-veterans-affairs/vets-website/pull/31588) is waiting on platform approval, once that is merged this ticket can be closed
